From b8d079afa75eda5ce99da8e8da54cd5c62e59222 Mon Sep 17 00:00:00 2001
Message-Id: <b8d079afa75eda5ce99da8e8da54cd5c62e59222.1379450584.git.jdenemar@redhat.com>
From: =?UTF-8?q?J=C3=A1n=20Tomko?= <jtomko@redhat.com>
Date: Thu, 12 Sep 2013 10:28:10 +0200
Subject: [PATCH] tests: files named '.*-invalid.xml' should fail validation

https://bugzilla.redhat.com/show_bug.cgi?id=1006710

Currently, using an invalid XML in tests fails, because
the schema test expects all of them to be valid.

Treat files with -invalid.xml suffix as invalid and expect
them to fail validation.
(cherry picked from commit 470d5c4654b512d101a8665754b13866925eeaa2)
---
 tests/schematestutils.sh | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/tests/schematestutils.sh b/tests/schematestutils.sh
index 4361221..e739b99 100644
--- a/tests/schematestutils.sh
+++ b/tests/schematestutils.sh
@@ -20,6 +20,18 @@ do
     result=`$cmd 2>&1`
     ret=$?
 
+    grep -- '-invalid.xml$' <<< "$xml" 2>&1 >/dev/null
+    invalid=$?
+
+    # per xmllint man page, the return codes for validation error
+    # are 3 and 4
+    if test $invalid -eq 0; then
+        if test $ret -eq 4 || test $ret -eq 3; then
+            ret=0
+        elif test $ret -eq 0; then
+            ret=3
+        fi
+    fi
     test_result $n $(basename $(dirname $xml))"/"$(basename $xml) $ret
     if test "$verbose" = "1" && test $ret != 0 ; then
         printf '%s\n' "$cmd" "$result"
-- 
1.8.3.2

